server/consumer.go

```diff
@@ -3313,31 +3310,36 @@ func (o *consumer) getNextMsg() (*jsPubMsg, uint64, error) {
 	// to avoid reflection.
 	if o.subjf != nil && len(o.subjf) > 1 {
 		sort.Slice(o.subjf, func(i, j int) bool {
+			if o.subjf[j].pmsg != nil && o.subjf[i].pmsg == nil {
```

Comment:
Checking for `pmsg` not being nil mainly prevents errors when two equal filter sequences (at the end of the stream) happen, and one has the last message.

Through all tests, it looks safe to skip checking if they are equal and just order them properly based `pmsg` absence or not.